deployment/docker-compose.yml

```diff
@@ -22,12 +22,14 @@ services:
   postgres:
     image: postgres
     container_name: postgres
+    env_file:
+      - .env
     ports:
-      - 5432:5432
+      - ${POSTGRES_PORT}:5432
     environment:
```

Comment:
Can you please explain what do you mean?
I'm a bit confused.
`POSTGRES_USER: ${POSTGRES_USER}`
here `POSTGRES_USER` - is a variable inside container and `${POSTGRES_USER}` - is the value we assign to internal container variable. In our case we specify `POSTGRES_USER` in env file, so to use it in compose file we write `${POSTGRES_USER}`.

Reply:
If I remove this section, `docker-compose up postgres` still works, that's how I concluded that this section is redundant. What's the difference between "a variable inside container" and "internal container variable"? AFAIU `env_file` and `environment` are used for essentially the same thing: setting variables inside the container. There seems to be no "internal container variable" or I don't understand what it means. My experiment confirms that.

From https://github.com/compose-spec/compose-spec/blob/master/spec.md:
> `env_file` adds environment variables to the container based on file content.

This description sounds like all variables from `env_file` will be set inside `postgres` container.
I just don't understand what would be the point of `env_file` if it worked differently. If we had to do `VAR: ${VAR}` for each variable, it would be really strange.

Reply:
oh, you are talking about both passing `env_file` and assigning values 
I've got it